blog: transient examples

bennn opened the pull request:
Blog post showing what transient (Shallow Typed Racket) does on a few example programs from the Racket mailing list. Spoiler: transient always does the right thing.

I'm planning to merge this tomorrow afternoon (2pm?) because I need to reference it for RacketCon this weekend.

(If you happen to comment after the merge, I'll update the post.)

ulysses4ever commented on blog/_src/posts/2020-10-15-shallow-mailing-list.md:
```diff
@@ -0,0 +1,414 @@
+    Date: 2020-10-15T13:32:12
+    Tags: typed racket, by Ben Greenman
+
+Several old questions from the Typed Racket mailing list have new and simple
```
Maybe here (or elsewhere) it’s a good time to establish a connection between transient and shallow. Otherwise you just switch one for another suddenly.

ulysses4ever commented on blog/_src/posts/2020-10-15-shallow-mailing-list.md:
```diff
@@ -0,0 +1,414 @@
+
+### What's going on?
+
+**Deep** tries to enforce the type with a Racket `or/c` contract, but or/c
```
```suggestion
**Deep** tries to enforce the type with a Racket `or/c` contract, but `or/c`
```

ulysses4ever commented on blog/_src/posts/2020-10-15-shallow-mailing-list.md:
```diff
@@ -0,0 +1,414 @@
+### What's going on?
+
+**Deep** tries to enforce the type with a Racket `or/c` contract, but or/c
+ does not allow two function contracts with the same arity because has to
```
This sentence needs more love, I think. It’s probably several sentences. See if you can put it more clear.

ulysses4ever commented on blog/_src/posts/2020-10-15-shallow-mailing-list.md:
```diff
@@ -0,0 +1,414 @@
+ does not allow two function contracts with the same arity because has to
+ pick one to apply at run-time, but cannot.
+
+Note, the "Type Checker" error message is much better than what or/c would
```
```suggestion
Note, the "Type Checker" error message is much better than what `or/c` would
```

ulysses4ever commented on blog/_src/posts/2020-10-15-shallow-mailing-list.md:
```diff
@@ -0,0 +1,414 @@
+
+### How's transient?
+
+**Shallow** simply checks that maybe-car accepts both arities inside the
```
```suggestion
**Shallow** simply checks that `maybe-car` accepts both arities inside the
```

ulysses4ever commented on blog/_src/posts/2020-10-15-shallow-mailing-list.md:
```diff
@@ -0,0 +1,414 @@
+### How's transient?
+
+**Shallow** simply checks that maybe-car accepts both arities inside the
+ case-> type.
```
```suggestion
 `case->` type.
```

ulysses4ever commented on blog/_src/posts/2020-10-15-shallow-mailing-list.md:
```diff
@@ -0,0 +1,414 @@
+
+**Shallow** simply checks that maybe-car accepts both arities inside the
+ case-> type.
+The code runs without error.
```
The/a error?

ulysses4ever commented on blog/_src/posts/2020-10-15-shallow-mailing-list.md:
```diff
@@ -0,0 +1,414 @@
+
+### What's going on?
+
+**Deep** enforces the type of `list-of-s2` with a listof contract, which
```
```suggestion
**Deep** enforces the type of `list-of-s2` with a `listof` contract, which
```

ulysses4ever commented on blog/_src/posts/2020-10-15-shallow-mailing-list.md:
```diff
@@ -0,0 +1,414 @@
+### What's going on?
+
+**Deep** enforces the type of `list-of-s2` with a listof contract, which
+ ends up making a copy of the original (empty) list as it traverses and
```
I only partially got the whole explanation. Maybe some references giving it a temporal angle of view could help, e.g. at which point the copy created. Maybe something else.

ulysses4ever commented on blog/_src/posts/2020-10-15-shallow-mailing-list.md:
```diff
@@ -0,0 +1,414 @@
+
+## Higher-Order Value as Any
+
+Original message : <https://groups.google.com/g/racket-users/c/cCQ6dRNybDg/m/CKXgX1PyBgAJ>
```
Maybe links to google-groups could be made prettier. Not sure.

bennn commented:
Thanks!

I left off backticks on purpose because the rendering looks ugly. (In the past I've used bold instead, but yesterday even that looked distracting.)

For google groups, I removed the `https://` prefix from the rendered version.

For all other comments, I updated the post.
